### PR TITLE
Added strings algorithms and it's test file #8

### LINKED ISSUE #8: all string algorithms implementation
**to solve this issue you must include some of the programs to make string algorithms easy usable**
like: - 
fizz buzz
delete_reoccurring_characters
strip_url_params
validate_coordinates
domain_extractor
merge_string_checker
add_binary
breaking_bad
decode_string
encode_decode
group_anagrams
int_to_roman
is_palindrome
license_number
make_sentence
multiply_strings
one_edit_distance
rabin_karp
reverse_string
reverse_vowel
reverse_words
roman_to_int
word_squares
and much more 
**note**:- not needed to implement every above-mentioned idea
 you can create your own if it's good.

### PR DESCRIPTION
**Pull request by Aryamanz29**
### Description

Added strings algorithms and a test file.

### Related Issue

Fixes #8 

### Type of Change:

- Feature

### How Has This Been Tested?

-Use python unittest framework for testing is file.

### Checklist:

- [x] I read the **CONTRIBUTING** document before making this pull request
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials
- [x] I have commented my code or provided relevant documentation, particularly in hard-to-understand areas
- [x] I have updated documentation accordingly
- [x] My changes generate no new warnings 
- [x] Existing tests passed

### CODE REVIEW

**Comment by Aryamanz29** [comment]
> ### Description
> Added strings algorithms and a test file.
> 
> ### Related Issue
> Fixes #8
> 
> ### Type of Change:
> * Feature
> 
> ### How Has This Been Tested?
> -Use python unittest framework for testing is file.
> 
> ### Checklist:
> * [x]  I read the **CONTRIBUTING** document before making this pull request
> * [x]  My code follows the style guidelines of this project
> * [x]  I have performed a self-review of my own code or materials
> * [x]  I have commented my code or provided relevant documentation, particularly in hard-to-understand areas
> * [x]  I have updated documentation accordingly
> * [x]  My changes generate no new warnings
> * [x] Existing tests passed

**Comment by devkapilbansal** [comment]
@Aryamanz29 write steps for testing

**Comment by Aryamanz29** [comment]
> @Aryamanz29 write steps for testing
Steps for testing on terminal/shell
1.>>python
2.Import unittest from the standard library
3.Run python string_tests.py

**Comment by Aryamanz29** [comment]
> > @Aryamanz29 write steps for testing


> Steps for testing on terminal/shell
> 1.>>python
> 2.Import unittest from the standard library
> 3.Run python string_tests.py
>

**Comment by Aryamanz29** [comment]
> > > @Aryamanz29 write steps for testing


> > Steps for testing on terminal/shell
> > 1.>>python
> > 2.Import unittest from the standard library
> > 3.Run python string_tests.py

File: dsalgo/string_algo.py [hunk 1]
```diff
@@ -0,0 +1,158 @@
+
```
Comment (devkapilbansal) [inline dsalgo/string_algo.py:1]: Why a blank line here??

File: dsalgo/string_algo.py [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+class Strings:
+
+# Implement the encode and decode methods.
```
Comment (devkapilbansal) [inline dsalgo/string_algo.py:4]: Your comment doesn't seem relevant and it's not explains anything

File: dsalgo/string_algo.py [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+     
+     def encode(self,strs):
+         """Encodes a list of strings to a single string.
```
Comment (devkapilbansal) [inline dsalgo/string_algo.py:7]: Can you please use docstrings here??

File: tests/string_tests.py [hunk 4]
```diff
@@ -0,0 +1,25 @@
+
+import unittest
+from dsalgo.string_algo import Strings
+class TestEncodeDecode(unittest.TestCase):
```
Comment (devkapilbansal) [inline tests/string_tests.py:4]: Add a blank line after import command

**Review by devkapilbansal — CHANGES_REQUESTED** [review]
Use docstrings everywhere

File: dsalgo/string_algo.py [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+        return the results
+        Complexity:
+        Time: O(n)
+        Space: O(n)
+        """
+        # Validate the input
+        if n < 1:
+            raise ValueError("n cannot be less than one")
+        if n is None:
+            raise TypeError("n cannot be None")
```
Comment (devkapilbansal) [inline dsalgo/string_algo.py:54]: Remove this function. @codesankalp @ravi5175 I don't think that it is valuable to us in any sense. What are your views?

File: dsalgo/string_algo.py [hunk 6]
```diff
@@ -0,0 +1,156 @@
+class Strings:
+    def encode(self, strs):
+        """
+        Encodes a list of strings to a single string.
+        params: List[str]
+        returns: str
+        """
```
Comment (devkapilbansal) [inline dsalgo/string_algo.py:7]: Not able to understand clearly what your function is doing by this. Please elaborate

File: dsalgo/string_algo.py [hunk 7]
```diff
@@ -0,0 +1,156 @@
...
+        res = ""
+        for string in strs.split():
+            res += str(len(string)) + ":" + string
+        return res
+
+    def decode(self, s):
+        """
+        Decodes a single string to a list of strings.
+        params: str
+        returns: List[str]
```
Comment (devkapilbansal) [inline dsalgo/string_algo.py:17]: Similar to elaborate

File: dsalgo/string_algo.py [hunk 8]
```diff
@@ -0,0 +1,156 @@
...
+        while i >= 0 or j >= 0 or c == 1:
+            if i >= 0:
+                c += ord(a[i]) - zero
+                i -= 1
+            if j >= 0:
+                c += ord(b[j]) - zero
+                j -= 1
+            s = chr(c % 2 + zero) + s
+            c //= 2
+        return s
```
Comment (devkapilbansal) [inline dsalgo/string_algo.py:93]: Isn't it a math algo?? Then why it is here

File: tests/string_tests.py (hunk 4 above)
Comment (devkapilbansal) [reply tests/string_tests.py:5]: only one is required

File: tests/string_tests.py [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+    def test_longest_palindrome(self):
+        self.assertAlmostEqual(
+            "asdadsa", self.st.longest_palindrome("dasdasdasdasdasdadsa"))
+
+    def test_roman(self):
+        self.assertAlmostEqual(9, self.st.roman_to_int("IX"))
+
+
+if __name__ == "__main__":
+    unittest.main()
```
Comment (devkapilbansal) [inline tests/string_tests.py:29]: Probably if we run this with unittest module, then we don't need this too

**Review by devkapilbansal — CHANGES_REQUESTED** [review]

File: dsalgo/string_algo.py (hunk 8 above)
Comment (Aryamanz29) [reply dsalgo/string_algo.py]: ohk I'll remove that one

**Comment by devkapilbansal** [comment]
Squash your commit and use proper commit guidelines

**Pull request closed without merging**